node-operators/block-producers/onboarding/account-setup.md

```diff
@@ -0,0 +1,152 @@
+
+```bash
+curl http://127.0.0.1:9944 -H \
+"Content-Type:application/json;charset=utf-8" -d \
+  '{
+    "jsonrpc":"2.0",
+    "id":1,
+    "method":"author_rotateKeys",
+    "params": []
+  }'
+```
+
+Your hex-encoded session keys will be printed to the terminal in the `"result"` field.
+
+--8<-- 'code/node-operators/block-producers/onboarding/account-setup/terminal/generate-session-keys.md'
+
+Make sure you write down your session keys; you'll need to map your session keys to your account in the next section.
+
+### Map Session Keys {: #map-session-keys }
+
+To perform the next step and map your session keys to your account, head to [Polkadot.js Apps](https://polkadot.js.org/apps/?rpc=wss://fraa-dancebox-rpc.a.dancebox.tanssi.network#/extrinsics){target=\_blank}, click on the **Developer** tab, select **Extrinsics** from the dropdown, and take the following steps:
+
+1. Select your account, which should be the same account that you previously self-delegated
+2. Select the **session** module and the **setKeys** extrinsic
+3. For **keys**, enter your session keys
+4. For **proof**, enter `0x`
+5. Click **Submit Transaction** and sign and send the transaction from your wallet
+
+![Create and submit an extrinsic to set session keys on Polkadot.js Apps](/images/node-operators/block-producers/onboarding/account-setup/setup-1.webp)
+
+Using the `session.keyOwner` method, you can verify that your session keys have been mapped to your account as expected. To do this on [Polkadot.js Apps](https://polkadot.js.org/apps/?rpc=wss://fraa-dancebox-rpc.a.dancebox.tanssi.network#/extrinsics){target=\_blank}, click on the **Developer** tab, select **Chain state** from the dropdown, and take the following steps:
+
+1. Select the **session** module and the **keyOwner** query
+2. Enter `nmbs` in the **SpCoreCryptoKeyTypeId** field
+3. For **Bytes**, enter your hex-encoded session keys
+4. Click the **+** button next to the extrinsic field
+5. The account associated with the session keys, which should be your account, will be displayed at the bottom of the page
+
+![Create and submit query to verify session keys on Polkadot.js Apps](/images/node-operators/block-producers/onboarding/account-setup/setup-2.webp)
+
+## Submit Self-Delegation {: #submit-self-delegation }
+
+The next step towards becoming eligible to produce blocks on Tanssi and Tanssi Appchains is to delegate to your own account. To do this, you'll be required to submit a minimum of {{ networks.dancebox.staking.min_self_del.dance }} DANCE tokens.
+
+However, you must be in the top {{ networks.dancebox.staking.top_eligible_candidates }} candidates by total stake to produce blocks. So, if the candidate at position {{ networks.dancebox.staking.top_eligible_candidates }} has a total stake of more than {{ networks.dancebox.staking.min_self_del.dance }} DANCE, you'll need to ensure that your total stake is greater than theirs.
```

Comment:
I know it's a variable, but are we sure it is a somewhat static number? The top eligible candidates are AFAIK: 
Dacenbox collators + (Total Appchains  * Collators per Appchain), now configured: 5 dancebox collators + (Total Appchains * 2).
The number of Appchains will be dynamic, having chains entering/leaving the block production protocol each session.

Reply:
I grabbed it from here: https://github.com/moondance-labs/tanssi/blob/master/runtime/dancebox/src/lib.rs#L1304

Reply:
I'm not totally sure, so I can find out

Reply:
Okay! Yes, that number just sets an upper bound to the list to limit memory usage. It's intentionally bigger than the actual maximum expected, and less than infinite haha

Reply:
It would be something like: "Block producers are assigned upon each session, requiring a total of [variable, currently 2] per Appchain, and a minimum of [variable, currently 3] to [variable, currently 5] to Tanssi. 
The block producers participating in the session will be picked from the list of candidates ordered by total stake,  until the total required is covered."

Reply:
AH ok gotcha! Thanks for clarifying! 🙂

Reply:
Ah ok gotcha! Thanks for clarifying! 🙂